keyboards/rhymestone/keymaps/default/keymap.c

```diff
@@ -0,0 +1,225 @@
+  //|---------+---------+---------+---------+---------+---------+---------+---------+---------+---------|
+       RGB_TOG,  RGB_HUI,  RGB_SAI,  RGB_VAI,  XXXXXXX,  KC_MS_L,  KC_MS_D,  KC_MS_U,  KC_MS_R,  KC_NLCK,
+  //|---------+---------+---------+---------+---------+---------+---------+---------+---------+---------|
+       RGB_MOD,  RGB_HUD,  RGB_SAD,  RGB_VAD,  XXXXXXX,  KC_BTN1,  KC_BTN2,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+  //`---------+---------+---------+---------+---------+---------+---------+---------+---------+---------'
+       _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______
+  //,---------------------------------------------------------------------------------------------------.
+  )
+};
+
+#define L_BASE _BASE
+#define L_LOWER (1<<_LOWER)
+#define L_RAISE (1<<_RAISE)
+#define L_ADJUST (1<<_ADJUST)
+#define L_ADJUST_TRI (L_ADJUST|L_RAISE|L_LOWER)
+
+#ifdef OLED_DRIVER_ENABLE
+#include <stdio.h>
+#include <string.h>
+
+typedef struct {
+  uint8_t state;
+  char name[8];
+}LAYER_DISPLAY_NAME;
+
+#define LAYER_DISPLAY_MAX 5
+const LAYER_DISPLAY_NAME layer_display_name[LAYER_DISPLAY_MAX] = {
+  {L_BASE, "Base"},
+  {L_BASE + 1, "Base"},
+  {L_LOWER, "Lower"},
+  {L_RAISE, "Raise"},
+  {L_ADJUST_TRI, "Adjust"}
+};
+
+static inline const char* get_leyer_status(void) {
+
+  for (uint8_t i = 0; i < LAYER_DISPLAY_MAX; ++i) {
+    if (layer_state == 0 && layer_display_name[i].state == default_layer_state) {
+
+      return layer_display_name[i].name;
+    } else if (layer_state != 0 && layer_display_name[i].state == layer_state) {
+
+      return layer_display_name[i].name;
+    }
+  }
+
+  return "?";
+}
+
+static char layer_status_buf[24] = "Layer state ready.\n";
+static inline void update_keymap_status(void) {
+
+  snprintf(layer_status_buf, sizeof(layer_status_buf) - 1, "OS:%s Layer:%s\n",
+    keymap_config.swap_lalt_lgui? "win" : "mac", get_leyer_status());
+}
+
+static inline void render_keymap_status(void) {
+
+  oled_write(layer_status_buf, false);
+}
+
+#define UPDATE_KEYMAP_STATUS() update_keymap_status()
+
+static inline void render_status(void) {
+
+  UPDATE_LED_STATUS();
+  RENDER_LED_STATUS();
+  render_keymap_status();
+  UPDATE_LOCK_STATUS();
+  RENDER_LOCK_STATUS();
+  RENDER_KEY_STATUS();
+}
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+
+  if (is_keyboard_master())
+    return OLED_ROTATION_180;  // flips the display 180 degrees if offhand
+  return rotation;
+}
+
+void oled_task_user(void) {
+
+  if (is_keyboard_master()) {
+    render_status();
+  } else {
+    render_logo();
+  }
+}
+
+#else
+
+#define UPDATE_KEYMAP_STATUS()
+
+#endif
+
+static inline void update_change_layer(bool pressed, uint8_t layer1, uint8_t layer2, uint8_t layer3) {
+
+  pressed ? layer_on(layer1) : layer_off(layer1);
+  IS_LAYER_ON(layer1) && IS_LAYER_ON(layer2) ? layer_on(layer3) : layer_off(layer3);
+}
```

Comment:
This isn't really needed.  There are two functions that handle this: 
https://docs.qmk.fm/#/ref_functions?id=olkb-tri-layers

Reply:
Okay, I fixed it.